supervisor/snapshots/snapshot.py

```diff
@@ -370,17 +371,49 @@ def _folder_save(name):
             try:
                 _LOGGER.info("Snapshot folder %s", name)
                 with SecureTarFile(tar_name, "w", key=self._key) as tar_file:
-                    tar_file.add(
-                        origin_dir,
-                        arcname=".",
-                        filter=exclude_filter(MAP_FOLDER_EXCLUDE.get(name, [])),
+                    arcname = "."
+                    tar_file.add(origin_dir, arcname, False)
+                    add_directory_contents(
+                        tar_file, origin_dir, MAP_FOLDER_EXCLUDE.get(name, []), arcname
```

Comment:
Don't make multiple calls. Extend the `add_directory_contents` to be the only called function